### PR TITLE
feat: deprovision environment

### PR DESCRIPTION
**Pull request by neurosnap**
feat: change name for environment
Screenshot 2023-08-09 at 3 29 32 PM

### CODE REVIEW

File: src/ui/pages/environment-detail-settings.tsx [hunk 1]
```diff
@@ -1,122 +1,245 @@
...
+  const env = useSelector((s: AppState) =>
+    selectEnvironmentById(s, { id: envId }),
   );
 
-  useEffect(() => {
-    setHandle(environment.handle);
-  }, [environment.id]);
+  const [confirm, setConfirm] = useState(false);
+  const [handle, setHandle] = useState<string>("");
+  const [errors, validate] = useValidator<string, typeof validators>(
```
Comment (madhuravius) [inline src/ui/pages/environment-detail-settings.tsx:50]: are you using the handle regex match as well (?) i swear we had one

Comment (madhuravius) [reply src/ui/pages/environment-detail-settings.tsx:50]: https://github.com/aptible/app-ui/blob/main/src/ui/shared/db/creator-form.tsx#L59 this one

Comment (neurosnap) [reply src/ui/pages/environment-detail-settings.tsx:50]: i didn't see any validators on the environment -- will double check

Comment (madhuravius) [reply src/ui/pages/environment-detail-settings.tsx:50]: figured it was the common handle validator we use everywhere else (some kind of reusable func/decorator thing)

**Comment by joeclayallday** [comment]
Tiny thing - we don't show a label of Confirm Name anywhere for these deprovision blocks, so maybe we remove that to pattern match.
Screen Shot 2023-08-09 at 7 18 09 PM

File: src/ui/pages/environment-detail-settings.tsx [hunk 2]
```diff
@@ -1,122 +1,245 @@
...
-              {isDeleting ? "Deleting..." : "Delete Environment"}
-            </Button>
-          </div>
+      </div>
+
+      <form onSubmit={onSubmit}>
+        <div>
+          This will permanently deprovision <strong>{env.handle}</strong>{" "}
+          environment. This action cannot be undone. If you want to proceed,
+          type the <strong>{env.handle}</strong> below to continue.
```
Comment (joeclayallday) [inline src/ui/pages/environment-detail-settings.tsx:195]: We should remove the word "the" here on line 195

Comment (neurosnap) [reply src/ui/pages/environment-detail-settings.tsx]: done, thanks

**Review by madhuravius — APPROVED** [review]

**Pull request merged**